pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -409,6 +420,10 @@ func (ctrl *Controller) syncKubeletConfig(key string) error {
 		glog.Infof("Applied KubeletConfig %v on MachineConfigPool %v", key, pool.Name)
 	}
 
+	ctrl.firstSync.Do(func() {
+		ctrl.readyFlag.Done()
+	})
```

Comment:
shouldn't this still be moved to the defer above to avoid not being called on errors (leaking the other goroutine at this point) or other successful paths?

Reply:
I don't think so, because we want to only say we're ready on success right?

Reply:
how do we spot that we're blocked on the wait() though? logs? is that a condition we must wait forever or do we need to error?

Reply:
if that's the case, cool